Project-II-Calculator/calculator/src/App.js

```diff
@@ -1,23 +1,146 @@
-import React from 'react';
-import './App.css';
-
-const App = () => {
-  return (
-    <div>
-      <h3>Welcome to React Calculator</h3>
-      <p>
-        We have given you a starter project. You'll want to build out your
-        components in their respective files, remove this code and replace it
-        with the proper components.
-      </p>
-      <p>
-        <strong>
-          Don't forget to `default export` your components and import them here
-          inside of this file in order to make them work.
-        </strong>
-      </p>
-    </div>
-  );
-};
+import React from "react";
+import "./App.css";
+import Display from "./components/DisplayComponents/CalculatorDisplay";
+import WhiteButtons from "./components/ButtonComponents/NumberButton";
+import RedButtons from "./components/ButtonComponents/ActionButton";
 
+const funcs = ["÷", "x", "-", "+", "="];
+const ints = [1, 2, 3, 4, 5, 6, 7, 8, 9];
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      displayVal: "0",
+      postOp: false,
+      storedVal: null,
+      storedOp: null
+    };
+  }
+
+  clear() {
+    let disp = this.state.displayVal;
+    if (this.state.postOp) {
+      this.setState({ displayVal: "confirm clear" });
+    }
+    if (!this.state.postOp || disp === "confirm clear") {
+      this.setState({
+        displayVal: "0",
+        postOp: false,
+        postEqual: false,
```

Comment:
it looks like `postEqual` is missing from the initial state, so some of these functions that are using it may return some unexpected behavior.